R/map_peptides_on_structure.R

```diff
@@ -0,0 +1,342 @@
+#' Maps peptides onto a PDB structure
+#'
+#' Peptides are mapped onto PDB structures based on their positions. This is accomplished by replacing the B-factor information in the structure file with values that allow
+#' highlighting peptides when the structure is coloured by B-factor. In addition to only highlighting peptides, a continuous variable such as fold changes associated with peptides
+#' can be mapped onto the structure as a colour gradient.
+#'
+#' @param peptide_data a data frame containing at least the input columns to this function. If a structure file is provided in the \code{structure_file} argument,
+#' this data frame should only contain peptide and structure information for the provided structure since otherwise the information cannot be mapped correctly to the file due
+#' to the missing PDB identifier.
+#' @param uniprot_id a character column in the \code{peptide_data} data frame that contains UniProt identifiers for a corresponding peptide.
+#' @param pdb_id a character column in the \code{peptide_data} data frame that contains PDB identifiers for structures in which a corresponding peptide is found. This
+#' column is not required if a structure file is provided in the \code{structure_file} argument.
+#' @param chain a character column in the \code{peptide_data} data frame that contains the name of the chain from the PDB structure in which the peptide is found. **Important:**
+#' please provide the author defined chain definitions for both ".cif" and ".pdb" files. When the output of the \code{find_peptide_in_pdb} function
+#' is used as the input for this function, this corresponds to the \code{auth_chain} column.
+#' @param start_in_pdb a numeric column in the \code{peptide_data} data frame that contains start positions for peptides in the corresponding PDB structure. This information
```

Comment:
Here you could mention that this can also be amino acid positions. For this, however, it should be possible to use the same column for start_in_pdb and end_in_pdb